# Analyzing Data

In [218]:
# Basic imports
from IPython.display import Markdown, display
import numpy as np
import pandas as pd
from scipy.stats import kurtosis, skew

# Data visualizations
import matplotlib.pyplot as plt
import seaborn as sns
import matplotlib.patches as mpatches
%matplotlib inline

In [220]:
# importing data
df = pd.read_csv("green_data/kc_house_data.csv")
df.head()

id        date     price  bedrooms  bathrooms  sqft_living  \
0  7399300360   5/24/2022  675000.0         4        1.0         1180   
1  8910500230  12/13/2021  920000.0         5        2.5         2770   
2  1180000275   9/29/2021  311000.0         6        2.0         2880   
3  1604601802  12/14/2021  775000.0         3        3.0         2160   
4  8562780790   8/24/2021  592500.0         2        2.0         1120   

   sqft_lot  floors waterfront greenbelt  ... sewer_system sqft_above  \
0      7140     1.0         NO        NO  ...       PUBLIC       1180   
1      6703     1.0         NO        NO  ...       PUBLIC       1570   
2      6156     1.0         NO        NO  ...       PUBLIC       1580   
3      1400     2.0         NO        NO  ...       PUBLIC       1090   
4       758     2.0         NO        NO  ...       PUBLIC       1120   

  sqft_basement sqft_garage sqft_patio yr_built  yr_renovated  \
0             0           0         40     1969             0   
1          1570           0        240     1950             0   
2          1580           0          0     1956             0   
3          1070         200        270     2010             0   
4           550         550         30     2012             0   

                                             address        lat       long  
0  2102 Southeast 21st Court, Renton, Washington ...  47.461975 -122.19052  
1  11231 Greenwood Avenue North, Seattle, Washing...  47.711525 -122.35591  
2  8504 South 113th Street, Seattle, Washington 9...  47.502045 -122.22520  
3  4079 Letitia Avenue South, Seattle, Washington...  47.566110 -122.29020  
4  2193 Northwest Talus Drive, Issaquah, Washingt...  47.532470 -122.07188  

[5 rows x 25 columns]

In [222]:
df.shape

(30155, 25)

In [223]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30155 entries, 0 to 30154
Data columns (total 25 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             30155 non-null  int64  
 1   date           30155 non-null  object 
 2   price          30155 non-null  float64
 3   bedrooms       30155 non-null  int64  
 4   bathrooms      30155 non-null  float64
 5   sqft_living    30155 non-null  int64  
 6   sqft_lot       30155 non-null  int64  
 7   floors         30155 non-null  float64
 8   waterfront     30155 non-null  object 
 9   greenbelt      30155 non-null  object 
 10  nuisance       30155 non-null  object 
 11  view           30155 non-null  object 
 12  condition      30155 non-null  object 
 13  grade          30155 non-null  object 
 14  heat_source    30123 non-null  object 
 15  sewer_system   30141 non-null  object 
 16  sqft_above     30155 non-null  int64  
 17  sqft_basement  30155 non-null  int64  
 18  sqft_g

Observations:
1.Not a ton of null values
2. mainly object, float and integer values

In [224]:
df.describe()

id         price      bedrooms     bathrooms   sqft_living  \
count  3.015500e+04  3.015500e+04  30155.000000  30155.000000  30155.000000   
mean   4.538104e+09  1.108536e+06      3.413530      2.334737   2112.424739   
std    2.882587e+09  8.963857e+05      0.981612      0.889556    974.044318   
min    1.000055e+06  2.736000e+04      0.000000      0.000000      3.000000   
25%    2.064175e+09  6.480000e+05      3.000000      2.000000   1420.000000   
50%    3.874011e+09  8.600000e+05      3.000000      2.500000   1920.000000   
75%    7.287100e+09  1.300000e+06      4.000000      3.000000   2619.500000   
max    9.904000e+09  3.075000e+07     13.000000     10.500000  15360.000000   

           sqft_lot        floors    sqft_above  sqft_basement   sqft_garage  \
count  3.015500e+04  30155.000000  30155.000000   30155.000000  30155.000000   
mean   1.672360e+04      1.543492   1809.826098     476.039396    330.211142   
std    6.038260e+04      0.567717    878.306131     579.631302    285.770536   
min    4.020000e+02      1.000000      2.000000       0.000000      0.000000   
25%    4.850000e+03      1.000000   1180.000000       0.000000      0.000000   
50%    7.480000e+03      1.500000   1560.000000       0.000000    400.000000   
75%    1.057900e+04      2.000000   2270.000000     940.000000    510.000000   
max    3.253932e+06      4.000000  12660.000000    8020.000000   3580.000000   

         sqft_patio      yr_built  yr_renovated           lat          long  
count  30155.000000  30155.000000  30155.000000  30155.000000  30155.000000  
mean     217.412038   1975.163953     90.922301     47.328076   -121.317397  
std      245.302792     32.067362    416.473038      1.434005      5.725475  
min        0.000000   1900.000000      0.000000     21.274240   -157.791480  
25%       40.000000   1953.000000      0.000000     47.405320   -122.326045  
50%      150.000000   1977.000000      0.000000     47.551380   -122.225585  
75%      320.000000   2003.000000      0.000000     47.669913   -122.116205  
max     4370.000000   2022.000000   2022.000000     64.824070    -70.074340

observations: 1. max bed is 13. This is either a mansion or small apartment 2. the largest number of baths is 10.5. is this in the same building?

# Importing/EDA for Crime Data

In [204]:
#importing crime data
crime_df = pd.read_csv("environment_data/crime.csv")
crime_df

case_number       incident_datetime nibrs_code  \
0       C21034525  10/29/2021 11:21:00 AM        120   
1       C21034462  10/28/2021 08:10:00 PM        290   
2       C21034548  10/29/2021 12:30:00 AM        240   
3       C21034576  10/28/2021 05:00:00 PM        13B   
4       C22012227  04/12/2022 08:36:00 PM        13B   
...           ...                     ...        ...   
63613   C23005210  02/13/2023 08:30:00 PM        240   
63614   C23005135  02/05/2023 12:00:00 AM        500   
63615   C23005124  02/13/2023 08:43:00 AM        500   
63616   C23005214  02/13/2023 09:22:00 PM        23C   
63617   C23005227  02/14/2023 12:12:00 AM        23F   

                                nibrs_code_name             block_address  \
0                                       Robbery    14900 Block 4TH AVE SW   
1      Destruction/Damage/Vandalism of Property   17600 Block 152ND PL SE   
2                           Motor Vehicle Theft     100 Block SW 112TH ST   
3                                Simple Assault  14400 Block 162ND AVE SE   
4                                Simple Assault    2800 Block NE 200TH ST   
...                                         ...                       ...   
63613                       Motor Vehicle Theft     13400 Block 3RD AVE S   
63614            Violation of No Contact Orders                       NaN   
63615            Violation of No Contact Orders                       NaN   
63616                               Shoplifting     14300 Block 1ST AVE S   
63617                  Theft From Motor Vehicle  17200 Block AURORA AVE N   

               city state    zip         reporting_area     district  \
0            BURIEN    WA  98166  Reporting Area N05051  District N5   
1            RENTON    WA  98058  Reporting Area F05007  District F5   
2           SEATTLE    WA  98146  Reporting Area K01012  District K1   
3            RENTON    WA  98059  Reporting Area F02003  District F2   
4         SHORELINE    WA  98155  Reporting Area A05004  District A5   
...             ...   ...    ...                    ...          ...   
63613        BURIEN    WA  98168  Reporting Area N02002  District N2   
63614  COTTAGE LAKE    WA  98072  Reporting Area B07002  District B7   
63615      FAIRWOOD    WA  98058  Reporting Area F05006  District F5   
63616        BURIEN    WA  98168  Reporting Area N02005  District N2   
63617     SHORELINE    WA  98133  Reporting Area A04002  District A4   

       precinct              created_at              updated_at  hour_of_day  \
0           4.0  10/29/2021 03:04:35 PM  10/29/2021 03:39:50 PM           11   
1           3.0  10/29/2021 02:49:33 PM  10/29/2021 02:50:04 PM           20   
2           4.0  10/29/2021 05:09:55 PM  10/29/2021 05:11:18 PM            0   
3           3.0  10/29/2021 07:00:11 PM  10/29/2021 07:05:25 PM           17   
4           5.0  04/13/2022 06:35:07 AM  04/13/2022 06:36:30 AM           20   
...         ...                     ...                     ...          ...   
63613       4.0  02/13/2023 11:19:16 PM  02/13/2023 11:20:09 PM           20   
63614       2.0  02/13/2023 12:02:45 PM  02/13/2023 12:09:40 PM            0   
63615       3.0  02/13/2023 10:15:07 AM  02/13/2023 10:20:42 AM            8   
63616       4.0  02/13/2023 10:22:48 PM  02/13/2023 10:24:00 PM           21   
63617       5.0  02/14/2023 02:39:45 AM  02/14/2023 03:01:07 AM            0   

       day_of_week           ID  
0                6  29495462116  
1                5  29495152983  
2                6  29497952486  
3                5  29500173040  
4                3  32749306796  
...            ...          ...  
63613            2  42216580302  
63614            1  42187118214  
63615            2  42186068781  
63616            2  42191625301  
63617            3  42282305074  

[63618 rows x 16 columns]

In [205]:
#BASIC CLEANING
#dropping unnecessary columns
crime_df1 = crime_df.drop(columns = ["case_number", "nibrs_code", "reporting_area", "district", "precinct", "created_at", "updated_at", "hour_of_day", "day_of_week", "ID"])
#grabbing the year from the incident_datetime column
crime_df1["Year"] = crime_df1["incident_datetime"].apply(lambda x: x[6:10])
#Changing the values of the city column to lower case with the first letter capitalized
crime_df1["city"] = crime_df1["city"].apply(lambda x: str(x).lower().title())
#renaming city column to City
crime_df1.rename(columns = {"city": "City"}, inplace = True)
#dropping duplicate incident reports
crime_df1.drop_duplicates(inplace = True)
#selecting incident reports only from 2021 and 2022
crime_df1 = crime_df1[(crime_df1["Year"] == "2021") | (crime_df1["Year"] == "2022")]
#dropping incdent_datetime column
crime_df1 = crime_df1.drop(columns = ["incident_datetime"])

In [206]:
#Created a list of cities in our original df
city_list = ["Seattle", "Renton", "Kent", "Auburn", "Bellevue", "Kirkland", "Federal Way", "Redmond", "Sammamish", "Maple Valley", "Shoreline", "Burien", "Woodinville", "Issaquah", "Covington", "North Bend", "Enumclaw", "Des Moines", "Mercer Island", "Kenmore", "Snoqualmie", "Black Diamond", "Bothell", "Seatac", "Duvall", "Lake Forest Park", "Vashon", "Newcastle", "Tukwila", "Carnation", "Normandy Park", "Fall City", "Pacific", "Ravensdale", "Clyde Hill", "Medina", "Algona", "Milton", "Skykomish", "Yarrow Point", "Beaux Arts", "Hunts Point", "Baring", "Preston", "Hobart", "Union Hill-Novelty Hill", "Snoqualmie Pass", "Lakeland North", "Fairwood", "Cottage Lake", "Ames Lake", "White Center", "Dilworth"]


In [207]:
#dropping incidents from cities that are not in our house sale data
crime_df1 = crime_df1[crime_df1['City'].isin(city_list)]

In [208]:
#dropping all NAN reports
crime_df1.dropna(subset = ["nibrs_code_name"])


nibrs_code_name  \
0                                          Robbery   
1         Destruction/Damage/Vandalism of Property   
2                              Motor Vehicle Theft   
3                                   Simple Assault   
4                                   Simple Assault   
...                                            ...   
63586                               Identity Theft   
63599                          Motor Vehicle Theft   
63600  Theft of Motor Vehicle Parts or Accessories   
63603                     Drug/Narcotic Violations   
63610                                   Bad Checks   

                      block_address       City state    zip  Year  
0            14900 Block 4TH AVE SW     Burien    WA  98166  2021  
1           17600 Block 152ND PL SE     Renton    WA  98058  2021  
2             100 Block SW 112TH ST    Seattle    WA  98146  2021  
3          14400 Block 162ND AVE SE     Renton    WA  98059  2021  
4            2800 Block NE 200TH ST  Shoreline    WA  98155  2022  
...                             ...        ...   ...    ...   ...  
63586  20800 Block SE MAY VALLEY RD   Issaquah    WA  98027  2022  
63599         28400 Block NE 4TH PL    Redmond    WA  98053  2021  
63600        14900 Block 4TH AVE SW     Burien    WA  98166  2022  
63603    S 160TH ST & AMBAUM BLVD S     Burien    WA    NaN  2022  
63610       4100 Block 205TH AVE SE  Sammamish    WA  98075  2022  

[31044 rows x 6 columns]

In [209]:
#grouping by city
crime_by_city = crime_df1.groupby(["City"]).count()

In [210]:
#reseting the index values
crime_by_city = crime_by_city[["nibrs_code_name"]].reset_index()

In [211]:
#renaming the incident report column to crime_number
crime_by_city.rename(columns = {"nibrs_code_name": "crime_number"}, inplace = True)

In [212]:
#dictionary with population numbers for each kings county city
pop_dict = {"Seattle": 762500, "Renton": 107500, "Kent": 137900, "Auburn": 78690, "Bellevue": 153900, "Kirkland": 93570, "Federal Way": 101800, "Redmond": 75270, "Sammamish": 68150, "Maple Valley": 28920, "Shoreline": 60320, "Burien": 52490, "Woodinville": 13450, "Issaquah": 40950, "Covington": 21200, "North Bend": 7915, "Enumclaw": 12910, "Des Moines": 33160, "Mercer Island": 25780, "Kenmore": 137900, "Snoqualmie": 14490, "Black Diamond": 6145, "Bothell": 29210, "Seatac": 31910, "Duvall": 8320, "Lake Forest Park": 13620, "Vashon": 11055, "Newcastle": 13560, "Tukwila": 22620, "Carnation": 2160, "Normandy Park": 6790, "Fall City": 2032, "Pacific": 7230, "Ravensdale": 390, "Clyde Hill": 3110, "Medina": 2915, "Algona": 3300, "Milton": 13560, "Skykomish": 165, "Yarrow Point": 1125, "Beaux Arts": 315, "Hunts Point": 460, "Baring": 267, "Preston": 523, "Hobart": 6932, "Union Hill-Novelty Hill": 22683, "Snoqualmie Pass": 390, "Lakeland North": 13879, "Fairwood": 10541, "Cottage Lake": 22965, "Ames Lake": 1486, "White Center": 16631, "Dilworth": 0.0}
#turning dictionary into a dataframe and transposing it
pop_df = pd.DataFrame(pop_dict, index = [0])
pop_df = pop_df.transpose()
pop_df = pop_df.sort_index()
pop_df = pop_df.reset_index()
pop_df.rename(columns = {"index": "City", 0: "Population"}, inplace = True)

In [213]:
#merging dataframe together
crime_per_capita = pd.merge(crime_by_city, pop_df, on = "City")
crime_per_capita

City  crime_number  Population
0                 Ames Lake             1      1486.0
1                    Auburn          1113     78690.0
2                    Baring             3       267.0
3                Beaux Arts             1       315.0
4                  Bellevue            64    153900.0
5             Black Diamond            24      6145.0
6                   Bothell            20     29210.0
7                    Burien          4128     52490.0
8                 Carnation           183      2160.0
9              Cottage Lake            37     22965.0
10                Covington           950     21200.0
11               Des Moines            32     33160.0
12                   Duvall            81      8320.0
13                 Enumclaw           238     12910.0
14                 Fairwood           122     10541.0
15                Fall City           177      2032.0
16              Federal Way           308    101800.0
17                   Hobart            23      6932.0
18                 Issaquah           523     40950.0
19                  Kenmore           600    137900.0
20                     Kent           927    137900.0
21                 Kirkland            11     93570.0
22         Lake Forest Park             7     13620.0
23           Lakeland North           107     13879.0
24             Maple Valley          1255     28920.0
25            Mercer Island             3     25780.0
26                   Milton             1     13560.0
27                Newcastle           284     13560.0
28            Normandy Park             3      6790.0
29               North Bend           186      7915.0
30                  Pacific             5      7230.0
31                  Preston            26       523.0
32               Ravensdale            82       390.0
33                  Redmond           478     75270.0
34                   Renton          1324    107500.0
35                Sammamish          1398     68150.0
36                   Seatac          2520     31910.0
37                  Seattle          8543    762500.0
38                Shoreline          3011     60320.0
39                Skykomish            21       165.0
40               Snoqualmie           210     14490.0
41          Snoqualmie Pass             3       390.0
42                  Tukwila           106     22620.0
43  Union Hill-Novelty Hill            39     22683.0
44                   Vashon           315     11055.0
45             White Center           341     16631.0
46              Woodinville          1210     13450.0

In [215]:
#creating a crime per capita column
crime_per_capita["crime_per_capita"] = crime_per_capita["crime_number"]/ crime_per_capita["Population"]
crime_per_capita

City  crime_number  Population  crime_per_capita
0                 Ames Lake             1      1486.0          0.000673
1                    Auburn          1113     78690.0          0.014144
2                    Baring             3       267.0          0.011236
3                Beaux Arts             1       315.0          0.003175
4                  Bellevue            64    153900.0          0.000416
5             Black Diamond            24      6145.0          0.003906
6                   Bothell            20     29210.0          0.000685
7                    Burien          4128     52490.0          0.078644
8                 Carnation           183      2160.0          0.084722
9              Cottage Lake            37     22965.0          0.001611
10                Covington           950     21200.0          0.044811
11               Des Moines            32     33160.0          0.000965
12                   Duvall            81      8320.0          0.009736
13                 Enumclaw           238     12910.0          0.018435
14                 Fairwood           122     10541.0          0.011574
15                Fall City           177      2032.0          0.087106
16              Federal Way           308    101800.0          0.003026
17                   Hobart            23      6932.0          0.003318
18                 Issaquah           523     40950.0          0.012772
19                  Kenmore           600    137900.0          0.004351
20                     Kent           927    137900.0          0.006722
21                 Kirkland            11     93570.0          0.000118
22         Lake Forest Park             7     13620.0          0.000514
23           Lakeland North           107     13879.0          0.007709
24             Maple Valley          1255     28920.0          0.043396
25            Mercer Island             3     25780.0          0.000116
26                   Milton             1     13560.0          0.000074
27                Newcastle           284     13560.0          0.020944
28            Normandy Park             3      6790.0          0.000442
29               North Bend           186      7915.0          0.023500
30                  Pacific             5      7230.0          0.000692
31                  Preston            26       523.0          0.049713
32               Ravensdale            82       390.0          0.210256
33                  Redmond           478     75270.0          0.006350
34                   Renton          1324    107500.0          0.012316
35                Sammamish          1398     68150.0          0.020514
36                   Seatac          2520     31910.0          0.078972
37                  Seattle          8543    762500.0          0.011204
38                Shoreline          3011     60320.0          0.049917
39                Skykomish            21       165.0          0.127273
40               Snoqualmie           210     14490.0          0.014493
41          Snoqualmie Pass             3       390.0          0.007692
42                  Tukwila           106     22620.0          0.004686
43  Union Hill-Novelty Hill            39     22683.0          0.001719
44                   Vashon           315     11055.0          0.028494
45             White Center           341     16631.0          0.020504
46              Woodinville          1210     13450.0          0.089963

In [108]:
crime_df1

incident_datetime                           nibrs_code_name  \
0      10/29/2021 11:21:00 AM                                   Robbery   
1      10/28/2021 08:10:00 PM  Destruction/Damage/Vandalism of Property   
2      10/29/2021 12:30:00 AM                       Motor Vehicle Theft   
3      10/28/2021 05:00:00 PM                            Simple Assault   
4      04/12/2022 08:36:00 PM                            Simple Assault   
...                       ...                                       ...   
63613  02/13/2023 08:30:00 PM                       Motor Vehicle Theft   
63614  02/05/2023 12:00:00 AM            Violation of No Contact Orders   
63615  02/13/2023 08:43:00 AM            Violation of No Contact Orders   
63616  02/13/2023 09:22:00 PM                               Shoplifting   
63617  02/14/2023 12:12:00 AM                  Theft From Motor Vehicle   

                  block_address          City state    zip  Year  
0        14900 Block 4TH AVE SW        Burien    WA  98166  2021  
1       17600 Block 152ND PL SE        Renton    WA  98058  2021  
2         100 Block SW 112TH ST       Seattle    WA  98146  2021  
3      14400 Block 162ND AVE SE        Renton    WA  98059  2021  
4        2800 Block NE 200TH ST     Shoreline    WA  98155  2022  
...                         ...           ...   ...    ...   ...  
63613     13400 Block 3RD AVE S        Burien    WA  98168  2023  
63614                       NaN  Cottage Lake    WA  98072  2023  
63615                       NaN      Fairwood    WA  98058  2023  
63616     14300 Block 1ST AVE S        Burien    WA  98168  2023  
63617  17200 Block AURORA AVE N     Shoreline    WA  98133  2023  

[63618 rows x 7 columns]

In [109]:
crime_df1

incident_datetime                           nibrs_code_name  \
0      10/29/2021 11:21:00 AM                                   Robbery   
1      10/28/2021 08:10:00 PM  Destruction/Damage/Vandalism of Property   
2      10/29/2021 12:30:00 AM                       Motor Vehicle Theft   
3      10/28/2021 05:00:00 PM                            Simple Assault   
4      04/12/2022 08:36:00 PM                            Simple Assault   
...                       ...                                       ...   
63613  02/13/2023 08:30:00 PM                       Motor Vehicle Theft   
63614  02/05/2023 12:00:00 AM            Violation of No Contact Orders   
63615  02/13/2023 08:43:00 AM            Violation of No Contact Orders   
63616  02/13/2023 09:22:00 PM                               Shoplifting   
63617  02/14/2023 12:12:00 AM                  Theft From Motor Vehicle   

                  block_address          City state    zip  Year  
0        14900 Block 4TH AVE SW        Burien    WA  98166  2021  
1       17600 Block 152ND PL SE        Renton    WA  98058  2021  
2         100 Block SW 112TH ST       Seattle    WA  98146  2021  
3      14400 Block 162ND AVE SE        Renton    WA  98059  2021  
4        2800 Block NE 200TH ST     Shoreline    WA  98155  2022  
...                         ...           ...   ...    ...   ...  
63613     13400 Block 3RD AVE S        Burien    WA  98168  2023  
63614                       NaN  Cottage Lake    WA  98072  2023  
63615                       NaN      Fairwood    WA  98058  2023  
63616     14300 Block 1ST AVE S        Burien    WA  98168  2023  
63617  17200 Block AURORA AVE N     Shoreline    WA  98133  2023  

[63618 rows x 7 columns]

In [110]:
crime_df1[(crime_df1["incident_datetime"] == "03/08/2020 12:00:00 AM") & (crime_df1["block_address"] == "2700 Block I ST NE")]

incident_datetime nibrs_code_name       block_address    City  \
30263  03/08/2020 12:00:00 AM  Identity Theft  2700 Block I ST NE  Auburn   
32226  03/08/2020 12:00:00 AM  Identity Theft  2700 Block I ST NE  Auburn   

      state    zip  Year  
30263    WA  98002  2020  
32226    WA  98002  2020

In [111]:
crime_df1.drop_duplicates(inplace = True)

In [112]:
crime_df1

incident_datetime                           nibrs_code_name  \
0      10/29/2021 11:21:00 AM                                   Robbery   
1      10/28/2021 08:10:00 PM  Destruction/Damage/Vandalism of Property   
2      10/29/2021 12:30:00 AM                       Motor Vehicle Theft   
3      10/28/2021 05:00:00 PM                            Simple Assault   
4      04/12/2022 08:36:00 PM                            Simple Assault   
...                       ...                                       ...   
63613  02/13/2023 08:30:00 PM                       Motor Vehicle Theft   
63614  02/05/2023 12:00:00 AM            Violation of No Contact Orders   
63615  02/13/2023 08:43:00 AM            Violation of No Contact Orders   
63616  02/13/2023 09:22:00 PM                               Shoplifting   
63617  02/14/2023 12:12:00 AM                  Theft From Motor Vehicle   

                  block_address          City state    zip  Year  
0        14900 Block 4TH AVE SW        Burien    WA  98166  2021  
1       17600 Block 152ND PL SE        Renton    WA  98058  2021  
2         100 Block SW 112TH ST       Seattle    WA  98146  2021  
3      14400 Block 162ND AVE SE        Renton    WA  98059  2021  
4        2800 Block NE 200TH ST     Shoreline    WA  98155  2022  
...                         ...           ...   ...    ...   ...  
63613     13400 Block 3RD AVE S        Burien    WA  98168  2023  
63614                       NaN  Cottage Lake    WA  98072  2023  
63615                       NaN      Fairwood    WA  98058  2023  
63616     14300 Block 1ST AVE S        Burien    WA  98168  2023  
63617  17200 Block AURORA AVE N     Shoreline    WA  98133  2023  

[63325 rows x 7 columns]

In [113]:
crime_df1 = crime_df1[(crime_df1["Year"] == "2021") | (crime_df1["Year"] == "2022")]
crime_df1["Year"].value_counts()
crime_df1.head()

incident_datetime                           nibrs_code_name  \
0  10/29/2021 11:21:00 AM                                   Robbery   
1  10/28/2021 08:10:00 PM  Destruction/Damage/Vandalism of Property   
2  10/29/2021 12:30:00 AM                       Motor Vehicle Theft   
3  10/28/2021 05:00:00 PM                            Simple Assault   
4  04/12/2022 08:36:00 PM                            Simple Assault   

              block_address       City state    zip  Year  
0    14900 Block 4TH AVE SW     Burien    WA  98166  2021  
1   17600 Block 152ND PL SE     Renton    WA  98058  2021  
2     100 Block SW 112TH ST    Seattle    WA  98146  2021  
3  14400 Block 162ND AVE SE     Renton    WA  98059  2021  
4    2800 Block NE 200TH ST  Shoreline    WA  98155  2022

In [114]:
crime_df1 = crime_df1.drop(columns = ["incident_datetime"])

In [226]:
crime_df1.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 31044 entries, 0 to 63610
Data columns (total 6 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   nibrs_code_name  31044 non-null  object
 1   block_address    29525 non-null  object
 2   City             31044 non-null  object
 3   state            31044 non-null  object
 4   zip              30328 non-null  object
 5   Year             31044 non-null  object
dtypes: object(6)
memory usage: 1.7+ MB


In [115]:
crime_df1

nibrs_code_name  \
0                                          Robbery   
1         Destruction/Damage/Vandalism of Property   
2                              Motor Vehicle Theft   
3                                   Simple Assault   
4                                   Simple Assault   
...                                            ...   
63597                     Theft From Motor Vehicle   
63599                          Motor Vehicle Theft   
63600  Theft of Motor Vehicle Parts or Accessories   
63603                     Drug/Narcotic Violations   
63610                                   Bad Checks   

                    block_address            City state    zip  Year  
0          14900 Block 4TH AVE SW          Burien    WA  98166  2021  
1         17600 Block 152ND PL SE          Renton    WA  98058  2021  
2           100 Block SW 112TH ST         Seattle    WA  98146  2021  
3        14400 Block 162ND AVE SE          Renton    WA  98059  2021  
4          2800 Block NE 200TH ST       Shoreline    WA  98155  2022  
...                           ...             ...   ...    ...   ...  
63597       2500 Block S 370TH PL  Lakeland South    WA  98003  2022  
63599       28400 Block NE 4TH PL         Redmond    WA  98053  2021  
63600      14900 Block 4TH AVE SW          Burien    WA  98166  2022  
63603  S 160TH ST & AMBAUM BLVD S          Burien    WA    NaN  2022  
63610     4100 Block 205TH AVE SE       Sammamish    WA  98075  2022  

[32113 rows x 6 columns]

In [116]:
crime_df1["City"].value_counts()

Seattle         8543
Burien          4128
Shoreline       3011
Seatac          2520
Sammamish       1398
                ... 
Lake Stevens       1
Monroe             1
Kittitas           1
Cle Elum           1
Clinton            1
Name: City, Length: 114, dtype: int64

In [185]:
city_list = ["Seattle", "Renton", "Kent", "Auburn", "Bellevue", "Kirkland", "Federal Way", "Redmond", "Sammamish", "Maple Valley", "Shoreline", "Burien", "Woodinville", "Issaquah", "Covington", "North Bend", "Enumclaw", "Des Moines", "Mercer Island", "Kenmore", "Snoqualmie", "Black Diamond", "Bothell", "Seatac", "Duvall", "Lake Forest Park", "Vashon", "Newcastle", "Tukwila", "Carnation", "Normandy Park", "Fall City", "Pacific", "Ravensdale", "Clyde Hill", "Medina", "Algona", "Milton", "Skykomish", "Yarrow Point", "Beaux Arts", "Hunts Point", "Baring", "Preston", "Hobart", "Union Hill-Novelty Hill", "Snoqualmie Pass", "Lakeland North", "Fairwood", "Cottage Lake", "Ames Lake", "White Center", "Dilworth"]

len(city_list)
   

53

In [118]:
crime_df1 = crime_df1[crime_df1['City'].isin(city_list)]

In [120]:
crime_df1.dropna(subset = ["nibrs_code_name"])

nibrs_code_name  \
0                                          Robbery   
1         Destruction/Damage/Vandalism of Property   
2                              Motor Vehicle Theft   
3                                   Simple Assault   
4                                   Simple Assault   
...                                            ...   
63586                               Identity Theft   
63599                          Motor Vehicle Theft   
63600  Theft of Motor Vehicle Parts or Accessories   
63603                     Drug/Narcotic Violations   
63610                                   Bad Checks   

                      block_address       City state    zip  Year  
0            14900 Block 4TH AVE SW     Burien    WA  98166  2021  
1           17600 Block 152ND PL SE     Renton    WA  98058  2021  
2             100 Block SW 112TH ST    Seattle    WA  98146  2021  
3          14400 Block 162ND AVE SE     Renton    WA  98059  2021  
4            2800 Block NE 200TH ST  Shoreline    WA  98155  2022  
...                             ...        ...   ...    ...   ...  
63586  20800 Block SE MAY VALLEY RD   Issaquah    WA  98027  2022  
63599         28400 Block NE 4TH PL    Redmond    WA  98053  2021  
63600        14900 Block 4TH AVE SW     Burien    WA  98166  2022  
63603    S 160TH ST & AMBAUM BLVD S     Burien    WA    NaN  2022  
63610       4100 Block 205TH AVE SE  Sammamish    WA  98075  2022  

[31044 rows x 6 columns]

In [129]:
crime_by_city = crime_df1.groupby(["City"]).count()

In [161]:
crime_by_city = crime_by_city[["nibrs_code_name"]].reset_index()

In [165]:
crime_by_city.rename(columns = {"nibrs_code_name": "crime_number"}, inplace = True)




In [178]:
crime_by_city

City  crime_number
0                 Ames Lake             1
1                    Auburn          1113
2                    Baring             3
3                Beaux Arts             1
4                  Bellevue            64
5             Black Diamond            24
6                   Bothell            20
7                    Burien          4128
8                 Carnation           183
9              Cottage Lake            37
10                Covington           950
11               Des Moines            32
12                   Duvall            81
13                 Enumclaw           238
14                 Fairwood           122
15                Fall City           177
16              Federal Way           308
17                   Hobart            23
18                 Issaquah           523
19                  Kenmore           600
20                     Kent           927
21                 Kirkland            11
22         Lake Forest Park             7
23           Lakeland North           107
24             Maple Valley          1255
25            Mercer Island             3
26                   Milton             1
27                Newcastle           284
28            Normandy Park             3
29               North Bend           186
30                  Pacific             5
31                  Preston            26
32               Ravensdale            82
33                  Redmond           478
34                   Renton          1324
35                Sammamish          1398
36                   Seatac          2520
37                  Seattle          8543
38                Shoreline          3011
39                Skykomish            21
40               Snoqualmie           210
41          Snoqualmie Pass             3
42                  Tukwila           106
43  Union Hill-Novelty Hill            39
44                   Vashon           315
45             White Center           341
46              Woodinville          1210

In [186]:
pop_dict = {"Seattle": 762500, "Renton": 107500, "Kent": 137900, "Auburn": 78690, "Bellevue": 153900, "Kirkland": 93570, "Federal Way": 101800, "Redmond": 75270, "Sammamish": 68150, "Maple Valley": 28920, "Shoreline": 60320, "Burien": 52490, "Woodinville": 13450, "Issaquah": 40950, "Covington": 21200, "North Bend": 7915, "Enumclaw": 12910, "Des Moines": 33160, "Mercer Island": 25780, "Kenmore": 137900, "Snoqualmie": 14490, "Black Diamond": 6145, "Bothell": 29210, "Seatac": 31910, "Duvall": 8320, "Lake Forest Park": 13620, "Vashon": 11055, "Newcastle": 13560, "Tukwila": 22620, "Carnation": 2160, "Normandy Park": 6790, "Fall City": 2032, "Pacific": 7230, "Ravensdale": 390, "Clyde Hill": 3110, "Medina": 2915, "Algona": 3300, "Milton": 13560, "Skykomish": 165, "Yarrow Point": 1125, "Beaux Arts": 315, "Hunts Point": 460, "Baring": 267, "Preston": 523, "Hobart": 6932, "Union Hill-Novelty Hill": 22683, "Snoqualmie Pass": 390, "Lakeland North": 13879, "Fairwood": 10541, "Cottage Lake": 22965, "Ames Lake": 1486, "White Center": 16631, "Dilworth": 0.0}



In [187]:
len(pop_dict)

53

In [188]:
pop_df = pd.DataFrame(pop_dict, index = [0])
pop_df = pop_df.transpose()

In [189]:
pop_df = pop_df.sort_index()

In [190]:
pop_df = pop_df.reset_index()
pop_df.rename(columns = {"index": "City", 0: "Population"}, inplace = True)

In [191]:
crime_per_capita = pd.merge(crime_by_city, pop_df, on = "City")

In [192]:
crime_per_capita

City  crime_number  Population
0                 Ames Lake             1      1486.0
1                    Auburn          1113     78690.0
2                    Baring             3       267.0
3                Beaux Arts             1       315.0
4                  Bellevue            64    153900.0
5             Black Diamond            24      6145.0
6                   Bothell            20     29210.0
7                    Burien          4128     52490.0
8                 Carnation           183      2160.0
9              Cottage Lake            37     22965.0
10                Covington           950     21200.0
11               Des Moines            32     33160.0
12                   Duvall            81      8320.0
13                 Enumclaw           238     12910.0
14                 Fairwood           122     10541.0
15                Fall City           177      2032.0
16              Federal Way           308    101800.0
17                   Hobart            23      6932.0
18                 Issaquah           523     40950.0
19                  Kenmore           600    137900.0
20                     Kent           927    137900.0
21                 Kirkland            11     93570.0
22         Lake Forest Park             7     13620.0
23           Lakeland North           107     13879.0
24             Maple Valley          1255     28920.0
25            Mercer Island             3     25780.0
26                   Milton             1     13560.0
27                Newcastle           284     13560.0
28            Normandy Park             3      6790.0
29               North Bend           186      7915.0
30                  Pacific             5      7230.0
31                  Preston            26       523.0
32               Ravensdale            82       390.0
33                  Redmond           478     75270.0
34                   Renton          1324    107500.0
35                Sammamish          1398     68150.0
36                   Seatac          2520     31910.0
37                  Seattle          8543    762500.0
38                Shoreline          3011     60320.0
39                Skykomish            21       165.0
40               Snoqualmie           210     14490.0
41          Snoqualmie Pass             3       390.0
42                  Tukwila           106     22620.0
43  Union Hill-Novelty Hill            39     22683.0
44                   Vashon           315     11055.0
45             White Center           341     16631.0
46              Woodinville          1210     13450.0

In [193]:
crime_per_capita["crime_per_capita"] = crime_per_capita["crime_number"]/ crime_per_capita["Population"]




In [194]:
crime_per_capita
#Note that a few towns might be missing

City  crime_number  Population  crime_per_capita
0                 Ames Lake             1      1486.0          0.000673
1                    Auburn          1113     78690.0          0.014144
2                    Baring             3       267.0          0.011236
3                Beaux Arts             1       315.0          0.003175
4                  Bellevue            64    153900.0          0.000416
5             Black Diamond            24      6145.0          0.003906
6                   Bothell            20     29210.0          0.000685
7                    Burien          4128     52490.0          0.078644
8                 Carnation           183      2160.0          0.084722
9              Cottage Lake            37     22965.0          0.001611
10                Covington           950     21200.0          0.044811
11               Des Moines            32     33160.0          0.000965
12                   Duvall            81      8320.0          0.009736
13                 Enumclaw           238     12910.0          0.018435
14                 Fairwood           122     10541.0          0.011574
15                Fall City           177      2032.0          0.087106
16              Federal Way           308    101800.0          0.003026
17                   Hobart            23      6932.0          0.003318
18                 Issaquah           523     40950.0          0.012772
19                  Kenmore           600    137900.0          0.004351
20                     Kent           927    137900.0          0.006722
21                 Kirkland            11     93570.0          0.000118
22         Lake Forest Park             7     13620.0          0.000514
23           Lakeland North           107     13879.0          0.007709
24             Maple Valley          1255     28920.0          0.043396
25            Mercer Island             3     25780.0          0.000116
26                   Milton             1     13560.0          0.000074
27                Newcastle           284     13560.0          0.020944
28            Normandy Park             3      6790.0          0.000442
29               North Bend           186      7915.0          0.023500
30                  Pacific             5      7230.0          0.000692
31                  Preston            26       523.0          0.049713
32               Ravensdale            82       390.0          0.210256
33                  Redmond           478     75270.0          0.006350
34                   Renton          1324    107500.0          0.012316
35                Sammamish          1398     68150.0          0.020514
36                   Seatac          2520     31910.0          0.078972
37                  Seattle          8543    762500.0          0.011204
38                Shoreline          3011     60320.0          0.049917
39                Skykomish            21       165.0          0.127273
40               Snoqualmie           210     14490.0          0.014493
41          Snoqualmie Pass             3       390.0          0.007692
42                  Tukwila           106     22620.0          0.004686
43  Union Hill-Novelty Hill            39     22683.0          0.001719
44                   Vashon           315     11055.0          0.028494
45             White Center           341     16631.0          0.020504
46              Woodinville          1210     13450.0          0.089963

# Importing/EDA for Park Data

In [202]:
# Importing/EDA for Crime Data
park_df = pd.read_csv("environment_data/park_area.csv")
park_df
park_df1 = park_df.drop(columns = ["OBJECTID", "KC_FAC_FID", "KCPARKFID", "SITETYPE", "SHAPE_Length"])
park_df1["OWNER"].value_counts()

City of Seattle                       409
King County Parks and Recreation      205
City of Bellevue                      113
City of Kent                           62
City of Kirkland                       45
City of Auburn                         39
City of Snoqualmie                     39
City of Federal Way                    34
City of Renton                         34
City of Redmond                        33
City of Issaquah                       29
City of Shoreline                      27
City of Mercer Island                  24
City of Burien                         23
City of Enumclaw                       22
City of Tukwila                        21
City of Des Moines                     21
Washington State Parks                 20
Vashon PRSD                            19
City of Newcastle                      19
City of Bothell                        17
City of Sammamish                      13
City of Duvall                         13
Port of Seattle                   

In [201]:
park_df1["MAINTD_BY"].value_counts()

City of Seattle                            414
King County Parks and Recreation           209
City of Bellevue                           112
City of Kent                                63
City of Kirkland                            45
City of Auburn                              42
City of Snoqualmie                          39
City of Federal Way                         35
City of Renton                              34
City of Redmond                             33
City of Issaquah                            29
City of Shoreline                           28
City of Mercer Island                       24
City of Enumclaw                            23
City of Burien                              23
City of Tukwila                             22
City of Des Moines                          21
Washington State Parks                      20
Vashon PRSD                                 19
City of Newcastle                           19
City of Bothell                             18
City of Samma